src/main/java/com/zendesk/maxwell/producer/HttpPostProducer.java

```diff
@@ -0,0 +1,115 @@
+package com.zendesk.maxwell.producer;
+
+import com.zendesk.maxwell.MaxwellContext;
+import com.zendesk.maxwell.row.RowMap;
+
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpResponseException;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.http.ByteArrayContent;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpBackOffUnsuccessfulResponseHandler;
+import com.google.api.client.util.ExponentialBackOff;
+
+import com.google.api.client.http.javanet.NetHttpTransport;
+import org.apache.commons.codec.binary.Hex;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import java.io.IOException;
+
+public class HttpPostProducer extends AbstractProducer {
+    static final String HTTP_HMAC_HEADER = "x-signature";
+    static final String HTTP_CONENT_TYPE_HEADER = "application/json";
+    static final String HMAC_ALG = "HmacSHA1";
```

Comment:
sure, at least there's a reference implementation (https://github.com/joyent/node-http-signature/blob/master/http_signing.md).  Would you mind contributing whatever signature verification code you end up writing too?  Since all of this is a bit non-standard it's best to have some example code. 

If you want to do it as an abstract class that's fine.  It's on my list somewhere to release maxwell as a .jar so that this kind of functionality can be added as user-supplied producers.